lib/client.js

```diff
@@ -120,6 +102,31 @@ Client.prototype.connect = function () {
   });
 };
 
+Client.prototype.setupBrokerProfiles = function (brokers) {
+  this.brokerProfiles = Object.create(null);
+  var self = this;
+
+  Object.keys(brokers).forEach(function (key, index) {
+    var brokerProfile = brokers[key];
+    var addr = brokerProfile.host + ':' + brokerProfile.port;
+    self.brokerProfiles[addr] = brokerProfile;
+    if (self.ssl) {
+      var sslAddress = brokerProfile.endpoints.find(function (endpoint) {
+        return url.parse(endpoint).protocol === 'ssl:';
+      });
+      if (!sslAddress) {
+        var err = new Error('No SSL kafka endpoint found');
```

Comment:
You probably should include more information about the specific broker that does not have an SSL endpoint. 

Also, does every broker need to have an SSL endpoint or do we need to simply find at least one broker that has an SSL endpoint? I'm worried that the `error` that you emit here will cause an exception (since `EventEmitter`, by default, will throw error if there is no listener for the `error` event) even if there are other valid brokers with SSL endpoint.